charts/flyte-core/templates/propeller/deployment.yaml

```diff
@@ -2,29 +2,48 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ template "flytepropeller.name" . }}
   namespace: {{ template "flyte.namespace" . }}
+  {{- if .Values.flytepropeller.manager }}
+  name: {{ template "flytepropeller-manager.name" . }}
+  labels: {{ include "flytepropeller-manager.labels" . | nindent 4 }}
+  {{- else }}
+  name: {{ template "flytepropeller.name" . }}
   labels: {{ include "flytepropeller.labels" . | nindent 4 }}
+  {{- end }}
 spec:
   replicas: {{ .Values.flytepropeller.replicaCount }}
   selector:
+    {{- if .Values.flytepropeller.manager }}
+    matchLabels: {{ include "flytepropeller-manager.selectorLabels" . | nindent 6 }}
+    {{- else }}
     matchLabels: {{ include "flytepropeller.selectorLabels" . | nindent 6 }}
+    {{- end }}
   template:
     metadata:
       annotations:
         configChecksum: {{ include (print .Template.BasePath "/propeller/configmap.yaml") . | sha256sum | trunc 63 | quote }}
         {{- with .Values.flytepropeller.podAnnotations }}
         {{- toYaml . | nindent 8 }}
         {{- end }}
+      {{- if .Values.flytepropeller.manager }}
+      labels: {{ include "flytepropeller-manager.labels" . | nindent 8 }}
+      {{- else }}
       labels: {{ include "flytepropeller.labels" . | nindent 8 }}
+      {{- end }}
     spec:
       securityContext:
         fsGroup: 65534
         runAsUser: 1001
         fsGroupChangePolicy: "Always"
       containers:
       - command:
+        {{- if .Values.flytepropeller.manager }}
+        - flytepropeller-manager
+        - --manager.pod-template-namespace
+        - {{ template "flyte.namespace" . }}
```

Comment:
Not a helm expert. Should this say `include` or `template` ?

Reply:
@evalsocket can probably provide more information on this. I just used the definition of the namespace I see elsewhere. I believe the 'template' is because of how templates are compiled [here](https://github.com/flyteorg/flyte/blob/0c4b9bd39a107a005227a763519ce5e49a97bfde/script/generate_helm.sh#L14) with the '-n flyte' flag.

Reply:
@EngHabu It looks like the "include" keyword is meant to be used in [Go pipelines](https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-include-function). I think we are using "template" correctly here?